code/go/internal/validator/folder_spec.go

```diff
@@ -7,71 +7,43 @@ package validator
 import (
 	"fmt"
 	"io/fs"
-	"io/ioutil"
 	"log"
 	"path"
 	"regexp"
 	"strings"
 
 	"github.com/pkg/errors"
-	"gopkg.in/yaml.v3"
 
 	ve "github.com/elastic/package-spec/code/go/internal/errors"
 	"github.com/elastic/package-spec/code/go/internal/spectypes"
 )
 
-const (
-	itemTypeFile   = "file"
-	itemTypeFolder = "folder"
+type validator struct {
+	spec       spectypes.ItemSpec
+	pkg        *Package
+	folderPath string
 
-	visibilityTypePublic  = "public"
-	visibilityTypePrivate = "private"
-)
-
-type folderSpec struct {
-	fs       fs.FS
-	specPath string
-	commonSpec
-
-	// These "validation-time" fields don't actually belong to the spec, storing
-	// them here for convenience by now.
 	totalSize     spectypes.FileSize
 	totalContents int
 }
 
-func (s *folderSpec) load(fs fs.FS, specPath string) error {
-	specFile, err := fs.Open(specPath)
-	if err != nil {
-		return errors.Wrap(err, "could not open folder specification file")
-	}
-	defer specFile.Close()
-
-	data, err := ioutil.ReadAll(specFile)
-	if err != nil {
-		return errors.Wrap(err, "could not read folder specification file")
-	}
-
-	var wrapper struct {
-		Spec *commonSpec `yaml:"spec"`
-	}
-	wrapper.Spec = &s.commonSpec
-	if err := yaml.Unmarshal(data, &wrapper); err != nil {
-		return errors.Wrap(err, "could not parse folder specification file")
-	}
+func newValidator(spec spectypes.ItemSpec, pkg *Package) *validator {
+	return newValidatorForPath(spec, pkg, ".")
+}
 
-	err = setDefaultValues(&s.commonSpec)
-	if err != nil {
-		return errors.Wrap(err, "could not set default values")
+func newValidatorForPath(spec spectypes.ItemSpec, pkg *Package, folderPath string) *validator {
+	return &validator{
+		spec:       spec,
+		pkg:        pkg,
+		folderPath: folderPath,
 	}
+}
 
-	propagateContentLimits(&s.commonSpec)
-
-	s.fs = fs
-	s.specPath = specPath
-	return nil
+func (v *validator) Validate() ve.ValidationErrors {
+	return v.validateSpec(v.spec, v.pkg, v.folderPath)
 }
 
-func (s *folderSpec) validate(pkg *Package, folderPath string) ve.ValidationErrors {
+func (v *validator) validateSpec(spec spectypes.ItemSpec, pkg *Package, folderPath string) ve.ValidationErrors {
```

Comment:
A `validator` struct has already spec, pkg and folderPath fields, cannot they be accesed through `v` instead of parameters ?
For instance [here](https://github.com/elastic/package-spec/pull/388/files?diff=split&w=0#diff-dbe281159d5f34e721f41774cf23c7914ebde34855a11a4126aa29f1f5217599R42-R43) are used as parameters the same fields of `v`:
```
func (v *validator) Validate() ve.ValidationErrors {
	return v.validateSpec(v.spec, v.pkg, v.folderPath)
```

Reply:
Good point, this comes from a previous version where I was reusing the validator for nested directories and calling it recursively with different parameters. I will refactor this for clarity.

Reply:
Done.